src/hooks/useFirebaseNotifications.ts

```diff
@@ -0,0 +1,31 @@
+import { useEffect } from 'react'
+
+import { FIREBASE_MESSAGING_SW_PATH, FIREBASE_OPTIONS } from '@/config/constants'
+import { initializeApp } from 'firebase/app'
+
+export const useFirebaseNotifications = (): void => {
+  // Register servicer worker
+  useEffect(() => {
+    if (typeof window === 'undefined' || !('serviceWorker' in navigator)) {
+      return
+    }
+
+    const hasFirebaseOptions = Object.values(FIREBASE_OPTIONS).every(Boolean)
+
+    if (!hasFirebaseOptions) {
+      return
+    }
+
+    initializeApp(FIREBASE_OPTIONS)
```

Comment:
Can this throw if some constants are configured wrong / change? Should we try / catch the initialization?

Reply:
Good point. I'll wrap it in a `try / catch`.

Reply:
I abstracted the logic and used it in both places where we initialise Firebase in 6f106769e7298409835617aa88d7d5a1d794f346.